### PR TITLE
Expand SRTrack to include track energy, length, and quality quantities

### PR DESCRIPTION
**Pull request by clarenc3**
Minimal commit to include more track information into the standard record track class. Should be relevant for all DUNE ND subdetectors, although this is specifically created with the TMS in mind.

### CODE REVIEW

**Comment by cjbackhouse** [comment]
We have got a bit of "short names, all lowecase" convention going on (yes, it's inconsistently applied, legacy reasons...) which you make these be `E` (or `recoE`), `len`, and `qual`.

File: duneanaobj/StandardRecord/SRTrack.h
Comment (chenel) [inline duneanaobj/StandardRecord/SRTrack.h:28]: just so it's clear to me, the reason you need this is because the tracks curve and so (end - start) isn't a good measure of length?

File: duneanaobj/StandardRecord/SRTrack.h
Comment (chenel) [inline duneanaobj/StandardRecord/SRTrack.h:27]: how does this differ from Evis above?

File: duneanaobj/StandardRecord/SRTrack.h
Comment (clarenc3) [reply duneanaobj/StandardRecord/SRTrack.h:28]: There is that, and that the TMS has a region of thin and thick steel plates. So 40cm of track in thin steel plate region is not equivalent to 40cm of track in thick steel plate region

File: duneanaobj/StandardRecord/SRTrack.h
Comment (clarenc3) [reply duneanaobj/StandardRecord/SRTrack.h:27]: It was meant to be the energy estimator using track length, whereas Evis I thought was the visible energy contained in the track?

Comment (chenel) [reply duneanaobj/StandardRecord/SRTrack.h:27]: Ah, I see.  In the analysis I was doing with CAFs the analagous ND-LAr estimator is not baked into the files but rather computed from the track lengths stored in the CAFs.  I don't suppose it matters too much---we can leave this here---but for my edification is there any reason that couldn't be done in this case?

**Review by chenel — APPROVED** [review]

**Comment by clarenc3** [comment]
> @clarenc3 do you think there's any other changes we'll need in order to fill TMS branches, or is this all? We should merge if this is everything.

This will unfortunately be an evolving project. From the TMS meeting yesterday we'll likely also have to save down cluster information, and it's likely the track information might be expanded too. 

But what we have now is likely to do the trick for now; I'll let you decide if that means it should be pulled into main or not

**Comment by chenel** [comment]
Ok.  Yeah, let's merge this one for now, and when you have a clearer idea about what else you'll need you can open a new PR.

**Pull request merged**
